invenio_accounts/models.py

```diff
@@ -62,6 +83,9 @@ class User(db.Model, Timestamp, UserMixin):
 
     id = db.Column(db.Integer, primary_key=True)
 
+    username = db.Column(db.String(255), unique=True)
```

Comment:
I think we have to do the username like in [UserProfiles](https://github.com/inveniosoftware/invenio-userprofiles/blob/master/invenio_userprofiles/models.py#L52-L75).

Basically we have two columns 1) case-preserving username used for display purposes and 2) a lower case version of a username with a unique constraint. This ensures we cannot have two usernames: ``lnielsen`` and ``LNIELSEN`` as they would both be the same username, but that we can preserve the case depending on how the user prefers to write it.